docs/source/concepts.rst

```diff
@@ -1,33 +1,45 @@
-.. _concepts:
+.. _tasks example: examples/tasks.ipynb
+.. _rpc example: examples/rpc.ipynb
+.. _broadcast example: examples/broadcast.ipynb
 
-Here we introduce the meaning of task, rpc, and broadcast and their usage scenarios, since the meaning of a the task in ``kiwipy`` differs slightly from the one in RabbitMQ's tutorial.
 
-Furthermore, we introduce the concepts of loop-dependent and thread-dependent communicators, which are objects that might be used directly by dependent libraries.
 
-***********************
+Concepts
+========
+
+Here we introduce the meaning of task, rpc, and broadcast and their usage scenarios, since the meaning of a the task in
+``kiwiPy`` differs slightly from the one in RabbitMQ's tutorial.
+
 Task
-***********************
+----
 
-The task in kiwipy is bidirectional and the return value can be obtained by the client, which is more like a simple rpc. Thus easy to be confused with rpc mode.
+Tasks are one to many messages.  This means that you sent out a task to a queue and there can be zero or more workers
+attached one of which will process the task when it is ready.  The result of the task can optionally be delivered to the
+sender.  See the `tasks example`_ to see this in action.
 
-*********
-Rpc
-*********
+RPC
+---
+
+A remote procedure is one-to-one.  This is used when you want to call a particular remote function/method and (usually)
+expect an immediate response. For example imagine asking a remote process to pause itself.  Here you would make a RPC
+and wait to get confirmation that it has, indeed, paused.  See the `rpc example`_ to see this in action.
 
-One-to-one with name identifier
 
-*********
 Broadcast
-*********
+---------
+
+One sender to zero or more consumers.  These are fire and forget calls that broadcast a message to anyone who is
+listening.  Consumers may optionally apply a filter to only receive messages that match some criteria.
+See the `broadcast example`_ to see this in action.
 
-One sender to multiple subscriber.
 
-****************
-Communicator
-****************
+Coroutines vs Threads
+=====================
 
-Loop communicator
-==================
+KiwiPy's RabbitMQ backend has both coroutine and threaded version of the Communicator.  If you're in a coroutine
+environment you may prefer to use the :class:`mincepy.rmq.RmqCommunicator` which uses coroutines and
```

Comment:
were you hungry when you wrote this? I agree a minced pie is delicious but I thought you were baking a kiwipy over here

Reply:
Haha, yeah, hungry for code.  Got too many pies in the oven and confused with my database library.